### ISSUE TITLE
SDL2: Fullscreen not backwards compatability

### ISSUE THREAD

**Issue by mcpalmer1980** [pygame/pygame#1713]
Games developed for pygame versions < 2.0 expect FULLSCREEN to change the resolution to equal the width and height requested by a call to pygame.setmode(). There were some issues with screen flicker and graphical artifacts when changing to fullscreen, which I hid by filling the screen black and delaying 200ms.

From what I understand, newer versions of pygame don't change the resolution but use a 'windowed fullscreen mode' instead. The pygame display surface does not scale to the display resolution in FULLSCREEN, however.

There is a SCALED flag that may be combined with FULLSCREEN when calling pygame.setmode(). This fills the screen as expected but the image is scaled using a nearest pixel algorithm that creates screen artifacts, especially in animations. Setting the SDL_RENDER_SCALE_QUALITY environment variable to 1 or 2 enables bi-linear or anisotropic filtering, respectively.

These issues are easily remedied and I'm happy with the options I've provided in my games. But there are many older pygame games that don't work properly and have nobody to update them. I would recommend defaulting to SCALED with bilinear filtering enabled by default, falling back to nearest pixel scaling only where bilinear hardware support is lacking.

**Comment by robertpfeiffer** [comment]
I don't think we should default to bilinear filtering, for technical reasons and because of the intended use cases of SCALED, but we can make the API more accessible and document it.
